// @storybook/cli/dist/_node-chunks/block-node-version-VZGNMRIA.js
import CJS_COMPAT_NODE_URL_zv09hezec4 from 'node:url';
import CJS_COMPAT_NODE_PATH_zv09hezec4 from 'node:path';
import CJS_COMPAT_NODE_MODULE_zv09hezec4 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_zv09hezec4.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_zv09hezec4.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_zv09hezec4.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  createBlocker
} from "./chunk-N7KQNM57.js";
import {
  require_semver
} from "./chunk-XFVW7Z75.js";
import {
  __toESM
} from "./chunk-OQELV2SQ.js";

// src/autoblock/block-node-version.ts
var import_semver = __toESM(require_semver(), 1);
var blocker = createBlocker({
  id: "minimumNode20",
  async check() {
    const nodeVersion = process.versions.node;
    if (nodeVersion && (0, import_semver.lt)(nodeVersion, "20.0.0")) {
      return { nodeVersion };
    }
    return false;
  },
  log(data) {
    return {
      title: "Node.js 20 support removed",
      message: `We've detected you're using Node.js v${data.nodeVersion}. Storybook needs Node.js 20 or higher.`,
      link: "https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#nodejs--20"
    };
  }
});
export {
  blocker
};
