// @storybook/cli/dist/_node-chunks/block-experimental-addon-test-CDE6XOIS.js
import CJS_COMPAT_NODE_URL_zv09hezec4 from 'node:url';
import CJS_COMPAT_NODE_PATH_zv09hezec4 from 'node:path';
import CJS_COMPAT_NODE_MODULE_zv09hezec4 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_zv09hezec4.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_zv09hezec4.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_zv09hezec4.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  createBlocker
} from "./chunk-N7KQNM57.js";
import {
  require_semver
} from "./chunk-XFVW7Z75.js";
import {
  __toESM
} from "./chunk-OQELV2SQ.js";

// src/autoblock/block-experimental-addon-test.ts
var import_semver = __toESM(require_semver(), 1);
import { dedent } from "ts-dedent";
var blocker = createBlocker({
  id: "experimentalAddonTestVitest",
  async check({ packageManager }) {
    const experimentalAddonTestVersion = await packageManager.getInstalledVersion(
      "@storybook/experimental-addon-test"
    );
    if (!experimentalAddonTestVersion) {
      return false;
    }
    const vitestVersion = await packageManager.getInstalledVersion("vitest");
    if (!vitestVersion) {
      return false;
    }
    return import_semver.default.lt(vitestVersion, "3.0.0");
  },
  log() {
    return {
      title: "Experimental Addon Test Vitest",
      message: dedent`
      @storybook/experimental-addon-test is being stabilized in Storybook 9.
      
      The addon will be renamed to @storybook/addon-vitest and as part of this stabilization, we have dropped support for Vitest 2.
      
      You have two options to proceed:
      1. Remove @storybook/experimental-addon-test if you don't need it
      2. Upgrade to Vitest 3 to continue using the addon
    `,
      link: "https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#experimental-addon-test-vitest"
    };
  }
});
export {
  blocker
};
