// @storybook/cli/dist/_node-chunks/block-major-version-ON443XC2.js
import CJS_COMPAT_NODE_URL_zv09hezec4 from 'node:url';
import CJS_COMPAT_NODE_PATH_zv09hezec4 from 'node:path';
import CJS_COMPAT_NODE_MODULE_zv09hezec4 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_zv09hezec4.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_zv09hezec4.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_zv09hezec4.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  createBlocker
} from "./chunk-N7KQNM57.js";
import {
  require_semver
} from "./chunk-XFVW7Z75.js";
import {
  __name,
  __toESM
} from "./chunk-OQELV2SQ.js";

// src/autoblock/block-major-version.ts
var import_semver = __toESM(require_semver(), 1);
import { versions } from "storybook/internal/common";
import { CLI_COLORS } from "storybook/internal/node-logger";
import { dedent } from "ts-dedent";
function validateVersionTransition(currentVersion, targetVersion) {
  if (!currentVersion || !targetVersion) {
    return "ok";
  }
  const current = (0, import_semver.parse)(currentVersion);
  const target = (0, import_semver.parse)(targetVersion);
  if (!current || !target) {
    return "ok";
  }
  if (current.major === 0 || target.major === 0) {
    return "ok";
  }
  if ((0, import_semver.gt)(currentVersion, targetVersion)) {
    return "downgrade";
  }
  const gap = target.major - current.major;
  return gap > 1 ? "gap-too-large" : "ok";
}
__name(validateVersionTransition, "validateVersionTransition");
var blocker = createBlocker({
  id: "major-version-gap",
  async check(options) {
    const { packageManager } = options;
    try {
      const currentStorybookVersion = packageManager.getAllDependencies().storybook;
      if (!currentStorybookVersion) {
        return false;
      }
      const target = versions.storybook;
      const result = validateVersionTransition(currentStorybookVersion, target);
      if (result === "ok") {
        return false;
      }
      return {
        currentVersion: currentStorybookVersion,
        reason: result
      };
    } catch (e) {
      return false;
    }
  },
  log(data) {
    const coercedVersion = (0, import_semver.coerce)(data.currentVersion);
    if (data.reason === "downgrade") {
      return {
        title: "Downgrade Not Supported",
        message: dedent`
          Your Storybook version (v${data.currentVersion}) is newer than the target release (v${versions.storybook}). Downgrading is not supported.
          `,
        link: "https://storybook.js.org/docs/releases/migration-guide?ref=upgrade"
      };
    }
    if (coercedVersion) {
      const currentMajor = (0, import_semver.major)(coercedVersion);
      const nextMajor = currentMajor + 1;
      return {
        title: "Major Version Gap Detected",
        message: dedent`
          Your Storybook version (v${data.currentVersion}) is more than one major version behind the target release (v${versions.storybook}). Please upgrade one major version at a time.
          
          You can upgrade to version ${nextMajor} by running:
          ${CLI_COLORS.info(`npx storybook@${nextMajor} upgrade`)}
        `,
        link: `https://storybook.js.org/docs/${nextMajor}/migration-guide?ref=upgrade`
      };
    }
    throw new Error("No message found");
  }
});
export {
  blocker,
  validateVersionTransition
};
