// anima-storybook-cli/dist/helpers/build.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupTempDirectory = exports.getBuildDir = exports.buildStorybook = exports.DEFAULT_BUILD_DIR = exports.DEFAULT_BUILD_COMMAND = void 0;
const child_process_1 = require("child_process");
const path_1 = __importDefault(require("path"));
const fs_extra_1 = __importDefault(require("fs-extra"));
exports.DEFAULT_BUILD_COMMAND = 'build-storybook';
exports.DEFAULT_BUILD_DIR = 'storybook-static';
const buildStorybook = (command, silent = false) => {
    return new Promise((resolve, reject) => {
        const buildCommand = command ?? exports.DEFAULT_BUILD_COMMAND;
        const result = (0, child_process_1.spawn)('npm', ['run', buildCommand], {
            cwd: process.cwd(),
            stdio: silent ? 'ignore' : 'inherit',
        });
        result.on('close', (code) => {
            if (code === 0) {
                resolve();
            }
            else {
                reject();
            }
        });
    });
};
exports.buildStorybook = buildStorybook;
const getBuildDir = (buildDir) => {
    const buildDirPath = buildDir ?? exports.DEFAULT_BUILD_DIR;
    if (path_1.default.isAbsolute(buildDirPath))
        return buildDirPath;
    return path_1.default.join(process.cwd(), buildDirPath);
};
exports.getBuildDir = getBuildDir;
const setupTempDirectory = (dir, { __DEV__ = false } = {}) => {
    const TEMP_DIR = path_1.default.join(process.cwd(), dir);
    if (__DEV__) {
        fs_extra_1.default.removeSync(TEMP_DIR);
        fs_extra_1.default.mkdirSync(TEMP_DIR);
    }
    return TEMP_DIR;
};
exports.setupTempDirectory = setupTempDirectory;
